app/controllers/api/v1/application_controller.rb

```diff
@@ -0,0 +1,2 @@
+class Api::V1::ApplicationController < Api::ApplicationController
```

Comment:
Not a bid deal but several ApplicationControllers in one application doesn't make sense. Why not something like BaseController?

Reply:
This way it looks more consistent to me. 
Each level has own application controller.